core/src/command/mod.rs

```diff
@@ -22,21 +22,9 @@ async fn create_link(
     url: String,
     state: tauri::State<'_, Cli>,
 ) -> Result<(), String> {
-    let url = match Url::parse(url.as_str()) {
-        Ok(val) => val,
-        Err(e) => return Err(e.to_string()),
-    };
-    let resource = arklib::id::ResourceId::compute_bytes(url.as_ref().as_bytes())
-        .expect("Error compute resource from url");
-    let domain = url.domain().expect("Url has no domain");
-    let path = format!("{}/{domain}-{}.link", &state.path, resource.crc32);
-    let mut link = Link::new(url, title, desc);
-    link.write_to_path(&state.path, &path, true)
-        .await
-        .expect("Custom error type needed");
+    super::create_link(title, desc, &url, &state.path).expect("Error creating link");
```

Comment:
Share the code between CLI and command to have only one create_link function to care

Reply:
Looks good to me in a debug.
I just reused it directly instead of the tauri re-export.